juju/juju
# [JUJU-4475] Add autocert cache domain service

The new domain service (simply a proxy to the key value autocert cache) is now used on the httpserver worker.

Also, as a bonus, we remove the entire autocert legacy state.

Nit: remove unused Logger interface on external controllers domain service and move to debug the logging of acme's `GetCertificate()` error, which is not an error on juju's side.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

The main scenario is to bootstrap a controller using `autocert-dns-name` set to a domain we control, on which we'll have to add the controllers IP address to the dns entry. Then we just `openssl` client request the domain to verify that the cert is signed by let's encrypt and not juju-ca.

This was done thanks to @tlm 's help.

```
juju bootstrap aws c --config logging-config="juju.worker.servicefactory.autocertcache=TRACE" --config autocert-dns-name=ipv6galore.tlm.dev

openssl s_client -connect ipv6galore.tlm.dev:443
CONNECTED(00000003)
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
...
```
